### PR TITLE
New build process

### PR DESCRIPTION
**Pull request by wonderhoss**
This adds a Makefile and configure script along with common build targets.
These replace the old Wercker build and prepare the repo for Prow.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,21 @@
-FROM golang:1.9 AS builder
+FROM golang:1.10.3 AS builder
```
Comment (mthssdrbrg) [inline Dockerfile]: Any reason for not going to v1.12?

File: Dockerfile [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
+COPY Gopkg.lock Gopkg.lock
+COPY Gopkg.toml Gopkg.toml
+
+RUN dep ensure --vendor-only
+
+COPY cmd/	cmd/
+
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o prom-rule-reloader github.com/pusher/prom-rule-reloader/cmd
+
+FROM alpine:3.8
```
Comment (mthssdrbrg) [inline Dockerfile]: Similar to my previous question, any reason for not going to 3.9? :)

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	  --enable=deadcode \
+	  --enable=gofmt \
+	  --enable=goimports \
+	  --skip-dirs=pkg/client/ \
+	  --deadline=120s \
+	  --tests ./...
+	@ echo
+
+vendor:
+	@ echo "$(GREEN)Puling dependencies$(NC)"
```
Comment (mthssdrbrg) [inline Makefile]: `s/Puling/Pulling`

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	@ echo
+
+.PHONY: check
+check: fmt lint vet test
+
+.PHONY: build
+build: clean $(BINARY)
+
+$(BINARY): fmt vet
+	CGO_ENABLED=0 $(GO) build -o $(BINARY) -ldflags="-X main.VERSION=${VERSION}" github.com/pusher/prom-rule-reloader/cmd
```
Comment (mthssdrbrg) [inline Makefile]: There's no `VERSION` variable specified in the `main` package, so we should either add that (could be added in `main.go`) or remove `-ldflags`.

Comment (mthssdrbrg) [reply Makefile]: Also just noticed that the Dockerfile doesn't include `-ldflags` when building the binary.

**Review by mthssdrbrg — CHANGES_REQUESTED** [review]
Looks good in general! Got a few questions and spotted a typo.

File: Dockerfile (hunk 1 above)
Comment (gargath) [reply Dockerfile]: Copied what faros is using, I think. Happy to try 1.12

File: Makefile (hunk 4 above)
Comment (gargath) [reply Makefile]: This is fixed in #6. I decided to make that a separate PR in the end. If we're merging them together I don't think there's a need to remove it here, wdyt?

Comment (mthssdrbrg) [reply Makefile]: I personally prefer that :)

**Review by mthssdrbrg — APPROVED** [review]
LGTM

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+.PHONY: vet
+vet: vendor
+	$(GO) vet ./cmd/...
+
+.PHONY: lint
+lint:
+	@ echo "$(GREEN)Linting code$(NC)"
+	$(LINTER) run --disable-all \
+	  --enable=vet \
```
Comment (JoelSpeed) [inline Makefile]: `golangci-lint` doesn't actually support `vet` and `vetshadow` but merges them into `govet` I believe.

It also ignores a bunch of lint rules that we have been using previously, see changes in https://github.com/pusher/faros/pull/125/files

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+	@IFS=","; tags=${TAGS}; for tag in $${tags}; do docker tag ${IMG}:${VERSION} ${IMG}:$${tag}; echo "$(GREEN)Tagged $(IMG):$(VERSION) as $${tag}$(NC)"; done
+
+PUSH_TAGS ?= ${VERSION}, latest
+.PHONY: docker-push
+docker-push: docker-build docker-tag
+	@IFS=","; tags=${PUSH_TAGS}; for tag in $${tags}; do docker push ${IMG}:$${tag}; echo "$(GREEN)Pushed $(IMG):$${tag}$(NC)"; done
+
+TAGS ?= latest
+.PHONY: docker-clean
+docker-clean:
```
Comment (JoelSpeed) [inline Makefile]: Useful, hadn't thought about doing something like this before

File: Makefile (hunk 5 above)
Comment (gargath) [reply Makefile]: I'll update the list. Thanks :)

**Review by JoelSpeed — APPROVED** [review]

**Pull request merged**